File: 0061-Rotate-Right/0061.java
class Solution {
  public ListNode rotateRight(ListNode head, int k) {
    if (k == 0 || head == null || head.next == null) {
      return head;
    }
    int n = 1;
    ListNode iter = head;
    while (iter.next != null) {
      iter = iter.next;
      n++;
    }
    int add = n - k % n;
    if (add == n) {
      return head;
    }
    iter.next = head;
    while (add-- > 0) {
      iter = iter.next;
    }
    ListNode ret = iter.next;
    iter.next = null;
    return ret;
  }
}

File: 0-100/0061-Rotate-Right/0061.py
class Solution:
  def rotateRight(self, head: ListNode, k: int) -> ListNode:
    if k == 0 or not head or not head.next:
      return head

    n = 1
    cur = head
    while cur.next:
      cur = cur.next
      n += 1

    if (add := n - k % n) == n:
      return head

    cur.next = head
    while add:
      cur = cur.next
      add -= 1

    ret = cur.next
    cur.next = None
    return ret